lib/search.js

```diff
@@ -8,8 +8,8 @@ class ExpectimaxTree {
 		this.rew_range = this.max_reward - this.min_reward;
 		this.numActions = agent.numActions;
 		this.samples = agent.samples;
-		this.utility = agent.utility;
 		this.gamma = agent.gamma;
+		this.agent = agent;
```

Comment:
as discussed, just pass in the discount function if possible. the planner shouldn't have access to the whole agent object.